modules/payment/payment.service.js

```diff
@@ -0,0 +1,85 @@
+const axios = require('axios')
+const CloudIpsp = require('cloudipsp-node-js-sdk')
+const { RSA_NO_PADDING } = require('constants')
+const crypto = require('crypto')
+
+
+class PaymentService {
+  genSignature (data, secret) {
+    const ordered = {}
+    Object.keys(data).sort().forEach(function (key) {
+      if (data[key] !== '' && key !== 'signature' && key !== 'response_signature_string') {
+        ordered[key] = data[key]
+      }
+    })
+    const signString = secret + '|' + Object.values(ordered).join('|')
+    return crypto.createHash('sha1').update(signString).digest('hex')
+  }
+
+    async getPaymentCheckout(data){  
+       const {
+            orderId = 'Unique12',
+            orderDesc = 'test order',
+            currency = 'UAH',
+            amount = 1
+        } = data 
+
+        const fondy = new CloudIpsp(
+            {
+              merchantId: process.env.PAYMENT_MERCHANT_ID,
+              secretKey: process.env.PAYMENT_SECRET
+            }
+          )
+          const requestData = {
+            order_id: orderId,
+            order_desc: orderDesc,
+            currency,
+            amount
+          }
+
+          const res = await fondy.Checkout(requestData)
+            .then(res => res)
+            .catch((error) => error)
+          
+          return {
+            paymentId: res.payment_id,
+            responseStatus: res.response_status,
+            checkoutUrl: res.checkout_url
+          }
+    }
+
+    async getPaymentRefund({
+      orderId = 'Unique12',
+      currency = 'UAH',
+      amount = 1,
+    }){  
+      const data = {
+        orderId,
+        currency,
+        amount,
+        merchantId: process.env.PAYMENT_MERCHANT_ID
+      }
+      const sig = this.genSignature(data, process.env.PAYMENT_SECRET)
+     
+      const res = await axios.post(
```

Comment:
What about error handling? Which kind of errors can be returned from fondi? Please, investigate it, cover the most common errors and return a valid and readable error message to the client